src/kernel.ts

```diff
@@ -1,1069 +1,803 @@
+
+    var future = new KernelFutureHandler(() => {
+      this._handlerMap.delete(msg.header.msgId);
+    });
+
+    this._handlerMap.set(msg.header.msgId, future);
+
+    return future;
+  }
+
+  /**
+   * Get kernel info.
+   *
+   * Returns a KernelFuture that will resolve to a `kernel_info_reply` message documented
+   * [here](http://ipython.org/ipython-doc/dev/development/messaging.html#kernel-info)
+   */
+  kernelInfo(): IKernelFuture {
+    return this.sendShellMessage("kernel_info_request", {});
+  }
+
+  /**
+   * Get info on an object.
+   *
+   * Returns a KernelFuture that will resolve to a `inspect_reply` message documented
+   * [here](http://ipython.org/ipython-doc/dev/development/messaging.html#object-information)
+   */
+  inspect(code: string, cursor_pos: number): IKernelFuture {
+    var content = {
+      code: code,
+      cursor_pos: cursor_pos,
+      detail_level: 0
+    };
+    return this.sendShellMessage("inspect_request", content);
+  }
+
+  /**
+   * Execute given code into kernel, returning a KernelFuture.
+   *
+   * @example
+   *
+   * The options object should contain the options for the execute
+   * call. Its default values are:
+   *
+   *      options = {
+   *        silent : true,
+   *        user_expressions : {},
+   *        allow_stdin : false,
+            store_history: false
+   *      }
+   *
+   */
+  execute(code: string, options?: { silent?: boolean; user_expressions?: any; allow_stdin?: boolean; store_history?: boolean; }): IKernelFuture {
+    var content = {
+      code: code,
+      silent: true,
+      store_history: false,
+      user_expressions: {},
+      allow_stdin: false
+    };
+    utils.extend(content, options);
+    return this.sendShellMessage("execute_request", content);
+  }
+
+  /**
+   * Request a code completion from the kernel.
+   *
+   * Returns a KernelFuture with will resolve to a `complete_reply` documented
+   * [here](http://ipython.org/ipython-doc/dev/development/messaging.html#complete)
+   */
+  complete(code: string, cursor_pos: number): IKernelFuture {
+    var content = {
+      code: code,
+      cursor_pos: cursor_pos
+    };
+    return this.sendShellMessage("complete_request", content);
+  }
+
+  /**
+   * Send an input reply message to the kernel.
+   *
+   * TODO: how to handle this?  Right now called by
+   * ./static/notebook/js/outputarea.js:827:        this.events.trigger('send_input_reply.Kernel', value);
+   *
+   * which has no referense to the session or the kernel
+   */
+  sendInputReply(input: any): string {
+    if (!this.isConnected) {
+      throw new Error("kernel is not connected");
+    }
+    var content = {
+      value: input
+    };
+    var msg = this._getMsg("input_reply", content);
+    msg.channel = 'stdin';
+    this._ws.send(serialize.serialize(msg));
+    return msg.header.msgId;
+  }
+
+  /**
+   * Get the Info Reply Message from the Kernel.
+   */
+  get infoReply(): any {
+    return this._infoReply()
```

Comment:
`this._infoReply` is not callable. Would have be caught by the compiler, but it's typed as `any`.